go/interpreter/lfvm/instructions.go

```diff
@@ -188,44 +175,26 @@ func opMcopy(c *context) error {
 
 	var destAddr = c.stack.pop()
 	var srcAddr = c.stack.pop()
-	var sizeU256 = c.stack.pop()
-
-	if sizeU256.IsZero() {
-		// zero size skips expansions although offset may be off-bounds
-		return nil
-	}
+	var size = c.stack.pop()
 
-	destOffset, destOverflow := destAddr.Uint64WithOverflow()
-	srcOffset, srcOverflow := srcAddr.Uint64WithOverflow()
-	if destOverflow || srcOverflow || !sizeU256.IsUint64() {
-		return errOverflow
-	}
-
-	size := sizeU256.Uint64()
-	price := tosca.Gas(3 * tosca.SizeInWords(size))
+	price := tosca.Gas(3 * tosca.SizeInWords(size.Uint64()))
```

Comment:
this could lead to deducting gas when size is greater than u256, for example if size is [1,1], and gas is >3, this `useGase` call would not fail when it should. I hope the example is clear enough.

Reply:
yes, this code crops overflows and charges only for the first 64 bit component. 
This is an example of error management shadowing other error returns. 
The reason why this is allowed is that `getSlice` is called and will perform the check nevertheless. Maybe it would be clear if getSlice is called before the word count cost.